### PR TITLE
[ADVA] Passing --disable-asm to econf for net-misc/rsync.

### PR DESCRIPTION
**Pull request by wikjas**
The optimization works properly only for x86 arch, so there is no need to use asm for cross platforms. 
Comparing the output of present compiler, use of asm has no impact on performance.

### CODE REVIEW

**Review by floppym — CHANGES_REQUESTED** [review]
Could you please elaborate on the purpose of this change? What problem does this solve?

Also, please ensure your submission conforms with [GLEP 66](https://www.gentoo.org/glep/glep-0066.html). Specifically:

- No merge commits, please rebase instead.
- Pay attention to the formatting the commit message. The first line should start with "net-misc/rsync:".

File: net-misc/rsync/rsync-3.2.3-r4.ebuild [hunk 1]
```diff
@@ -81,7 +81,7 @@ src_configure() {
 		myeconfargs+=( --disable-simd )
 	fi
 
-	econf "${myeconfargs[@]}"
+	econf --disable-asm "${myeconfargs[@]}"
```
[the remark this answers is not in the document]
Comment (thesamesam) [reply net-misc/rsync/rsync-3.2.3-r4.ebuild:84]: If we need to do this, I'd like to conditionally apply it in the cross case (inherit toolchain-funcs if it's not already and use a function from it to check) unless we have a very good reason.

Things like this tend to sit there forever so we may end up disabling optimisations unnecessarily in the future when it's useful on e.g. native ARM or whatever.

**Comment by floppym** [comment]
If this meant to fix cross-compiling, perhaps we should backport this commit instead:

https://github.com/WayneD/rsync/commit/7eb59a9152a2ace7bc7858e9915c671b3ab54344

**Comment by floppym** [comment]
It looks like the SIMD check was fixed upstream as well, so we could drop `--disable-simd` from the ebuild.

https://github.com/WayneD/rsync/commit/b7fab6f285ff0ff3816b109a8c3131b6ded0b484

**Pull request closed without merging**